core: move option results to the World class instead of MultiWorld (#993)
ArchipelagoMW/Archipelago: docs/world api.md

### Player Options

Players provide customized settings for their World in the form of yamls.

> Those are accessible through `self.world.options["<option_name>"]`. A dict

Reviewer: The documentation being replaced specifies the player. Shouldn't the replacement also specify the player?

class MyGameWorld(World):
    """Insert description of the world/game here."""
    game = "My Game"  # name of the game/world

>     options_dataclass = MyGameOptions # options the player can set

Reviewer: ```suggestion
    options_dataclass = MyGameOptions  # options the player can set
```
I think pep 8 calls for 2 spaces before a comment.

```python
def generate_early(self) -> None:
    # read player settings to world instance

>     self.final_boss_hp = self.world.options["final_boss_hp"].value

Reviewer: similar to previous comment: How does it know which player's `final_boss_hp` to get the value from?

Author: These dictionaries exist per `World` so this dictionary is all of the options that your current world's player chose.

Reviewer: What is the `self` that has a `world` on it? This doesn't include the PR that changes `world` to `multiworld`, right?

I think that's a `MultiWorld` that the documentation is referencing.

Author: it's a typo :)

    generate_mod(src, out_file, data)
```

### Slot Data

If the game client needs to know information about the generated seed, a preferred method of transferring the data

> is through the slot data. This can be filled in the `fill_slot_data` method on your world. This can be used to fill any
> data you need to know client side, and should return a `Dict[Any]`, but should be limited to only data that is
> absolutely necessary to not waste data. Slot data is sent to your client once it has successfully
> [connected.](network%20protocol.md#connected) If you need to know information about locations in your world, instead

Reviewer 2: ```suggestion
is through the slot data. This can be filled from the `fill_slot_data` method of your world by returing a `Dict[Any]`,
but should be limited to data that is absolutely necessary to not waste resources. Slot data is sent to your client once
it has successfully [connected](network%20protocol.md#connected).
If you need to know information about locations in your world, instead
```

Author: returing

Author: it's technically `Dict[Any, Any]` we just usually use `Dict[str, Any]`

Reviewer 2: fixed the suggestion. Can't be [Any, Any] because of json

Reviewer: The slot data needs to be JSON serializable.
The typing of JSON data has been discussed at length in other places https://github.com/python/typing/issues/182
Because of its complexity, I think it would be better to not mention the type and just mention that it needs to be JSON serializable.
